dashboard/src/contexts/DictionaryContextProvider.tsx

```diff
@@ -0,0 +1,77 @@
+'use client';
+
+import { createContext, useContext, useState, useCallback } from 'react';
+import { BADictionary, RawDictionary, loadDictionary, SupportedLanguages } from '@/dictionaries/dictionaries';
+
+interface DictionaryContextValue {
+  dictionary: BADictionary;
+  currentLanguage: SupportedLanguages;
+  changeLanguage: (language: SupportedLanguages) => Promise<void>;
+  isLoading: boolean;
+}
+
+const DictionaryContext = createContext<DictionaryContextValue | null>(null);
+
+type DictionaryProviderProps = {
+  dictionary: RawDictionary;
+  initialLanguage: SupportedLanguages;
+  children: React.ReactNode;
+};
+
+function addTFunction(dict: RawDictionary): BADictionary {
+  return {
+    ...dict,
+    t: (key: string) => {
+      return key.split('.').reduce((current: any, k) => current?.[k], dict) || key;
+    },
+  };
+}
+
+export default function DictionaryProvider({
+  dictionary: initialDictionary,
+  initialLanguage,
+  children,
+}: DictionaryProviderProps) {
+  const [dictionary, setDictionary] = useState(() => addTFunction(initialDictionary));
+  const [currentLanguage, setCurrentLanguage] = useState(initialLanguage);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const changeLanguage = useCallback(
+    async (language: SupportedLanguages) => {
+      if (language === currentLanguage) return;
+
+      setIsLoading(true);
+      try {
+        const newDictionary = await loadDictionary(language);
+        setDictionary(newDictionary);
+        setCurrentLanguage(language);
+      } catch (error) {
+        console.error('Failed to load dictionary:', error);
+      } finally {
+        setIsLoading(false);
+      }
+    },
+    [currentLanguage],
+  );
```

Comment:
Suggestion: Consider using `useSuspense` instead of manual loading state management

Reply:
Whops, meant `useTransition`. This is fine as is though 😄